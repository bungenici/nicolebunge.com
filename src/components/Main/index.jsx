import classNames from 'classnames';
import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from './style.module.css';

class Main extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
  };

  static defaultProps = {
    children: null,
    className: '',
  };

  render() {
    const { children, className, ...otherProps } = this.props;

    return (
      <main className={classNames(className, styles.main)} role="main" {...otherProps}>
        {children}
      </main>
    );
  }
}

export default Main;
