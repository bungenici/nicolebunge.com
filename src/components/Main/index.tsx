import classNames from 'classnames';
import React, { HTMLAttributes } from 'react';
import * as styles from './style.module.css';

export type MainProps = HTMLAttributes<HTMLDivElement>;

const Main: React.FC<MainProps> = (props) => {
  const { className, ...otherProps } = props;

  return <main className={classNames(className, styles.main)} role="main" {...otherProps} />;
};

export default Main;
