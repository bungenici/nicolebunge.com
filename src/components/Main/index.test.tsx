/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Main from '.';

describe('Main component', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Main />);

    expect(asFragment()).toMatchSnapshot();
  });
});
