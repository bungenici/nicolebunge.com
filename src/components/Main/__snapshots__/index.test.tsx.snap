// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders correctly 1`] = `
<DocumentFragment>
  <main
    class="main"
    role="main"
  />
</DocumentFragment>
`;
